**This is a peer comparison project. Suppose that you are working for Samsung in customer experience management team. The idea is to regularly monitor the customer satisfaction levels and peer company moves. The competitor company is Apple. The objective is to test two main hypothesis**

**a.The Samsung Average customer satisfaction score is minimum 75%.**

**b.The overall average satisfaction score of Samsung is same as Apple. There is no significant difference in the satisfaction
scores**


**It might be possible that both hypothesis are correct, one of them is correct or both of them are wrong. Perform the
relevant testing to verify these assumptions**

**The data is collected for 100 Samsung customers and 100 Apple customers. Their satisfaction scores are recorded. The sample represent the data and it is unbiased.**

In [1]:
import pandas as pd
import numpy as np

In [2]:
mobile_data=pd.read_csv('Testing_Of_Hypothesis.csv')
mobile_data

Sr No S_Cust_id  Samsung Score Apple_Cust_id  Apple_Score
0       1   S336499             94       A336525           82
1       2    S73004             99        A73011           83
2       3    S79537             74        A79538           88
3       4   S769168             74       A769222           80
4       5   S609791             83       A609801           74
5       6   S394037             87       A394057           82
6       7   S514183             89       A514215           88
7       8   S433504             83       A433564           92
8       9   S779836             75       A779861           69
9      10   S591911             74       A591964           64
10     11   S752225             80       A752245          100
11     12   S475483             86       A475484          100
12     13    S71938             87        A71953          100
13     14   S795414             89       A795427           70
14     15   S774214             99       A774256           94
15     16   S292726             88       A292752           67
16     17   S905753             78       A905761          100
17     18   S498459             97       A498474          100
18     19   S138232            100       A138268          100
19     20   S495322             98       A495338           87
20     21   S531256            100       A531282          100
21     22   S295462             81       A295508           89
22     23   S930105             69       A930118           99
23     24   S391026             69       A391065           69
24     25   S776161            100       A776189           80
25     26   S286774             96       A286795           95
26     27   S574618             73       A574667           93
27     28   S812219             84       A812248           98
28     29   S700090             67       A700111           80
29     30   S755713             98       A755714          100
..    ...       ...            ...           ...          ...
70     71   S314974             91       A314997           70
71     72    S29271             63        A29299          100
72     73   S729487            100       A729499           92
73     74   S825549             85       A825555           92
74     75   S362221             76       A362253           94
75     76   S818984             90       A819032           74
76     77   S205848            100       A205867           83
77     78   S742475             97       A742510           92
78     79    S29989             72        A30020           88
79     80   S149026            100       A149053           78
80     81   S117061             58       A117093           81
81     82   S239110            100       A239168           86
82     83   S756688             78       A756746          100
83     84   S642872             83       A642912           72
84     85   S298023             69       A298075           88
85     86   S261101             92       A261137           80
86     87   S182132             70       A182136           94
87     88    S67445             77        A67468          100
88     89   S287546             94       A287575           85
89     90   S826304             75       A826333           79
90     91   S960550             84       A960551           89
91     92   S740248             88       A740295           89
92     93   S487788             63       A487838           87
93     94    S40923             91        A40953           89
94     95   S509290             73       A509298           77
95     96   S285617             77       A285629           87
96     97   S324135             85       A324151           82
97     98   S292076             83       A292091           88
98     99   S936528             82       A936566           91
99    100   S489010            100       A489031           95

[100 rows x 5 columns]

H0: The Samsung Average customer satisfaction score = 75%
H1: The Samsung Average customer satisfaction score > 75%

In [3]:
mobile_data.columns

Index(['Sr No', 'S_Cust_id', 'Samsung Score', 'Apple_Cust_id', 'Apple_Score'], dtype='object')

In [4]:
mobile_data["Samsung Score"].mean()

84.9

In [5]:
mobile_data["Samsung Score"].std()

11.84453161982057

In [6]:
import scipy.stats as stats

In [7]:
stats.ttest_1samp(a=mobile_data["Samsung Score"],popmean=81)

Ttest_1sampResult(statistic=3.2926586927876222, pvalue=0.0013767729270006148)

P value is less than .05%. So rejecting the null hypothesis.
Avergae value is 84.9 which is higher than 75, so accepting the alternative hypothesis

## The overall average satisfaction score of Samsung is same as that of Apple

### HO: Average satisfaction score of Samsung == Apple
### HA: Average satisfaction score of Samsung != Apple

In [8]:
print((mobile_data["Samsung Score"].std())**2)
print((mobile_data["Apple_Score"].std())**2)

140.2929292929293
92.70090909090905


In [9]:
print(mobile_data["Samsung Score"].mean())
print(mobile_data["Apple_Score"].mean())

84.9
87.81


In [10]:
stats.ttest_ind(a=mobile_data['Samsung Score'], b=mobile_data['Apple_Score'],equal_var=False)

Ttest_indResult(statistic=-1.9064295935043474, pvalue=0.05810311766331876)

In [11]:
stats.ttest_ind(a=mobile_data['Samsung Score'], b=mobile_data['Apple_Score'],equal_var=True)

Ttest_indResult(statistic=-1.9064295935043472, pvalue=0.058042614444118075)

In [12]:
stats.f_oneway(mobile_data['Samsung Score'],mobile_data['Apple_Score'])

F_onewayResult(statistic=3.6344737949891592, pvalue=0.05804261444411817)

P- value is greater than 0.05 and the sample size is 100, we cannot reject null hypotheses. Therefore, overall average satisfaction score of samsung is equal to that of apple.